### DISCUSSION TITLE
How to get tls version for the httpx Client request

### DISCUSSION THREAD

**Discussion by heysarthak** [encode/httpx#2745] [Q&A]
i am following the below method

`conn = httpx.Client(http2=True,http1=False,verify=ssl_contexts)`
 `resp = conn.request("POST","https://{}:{}".format(serviceName,ip_port),data=data)`

How can i get the tls version of the given request?

**Comment by lovelydinosaur** [comment]
You might want to look at using [the `"trace"` extension](https://www.encode.io/httpcore/extensions/#trace) which will expose the `SSLContext` in the `"connection.start_tls.started"` event.

```python
def log(event_name, info):
    print(event_name, info)

resp = conn.request("POST", "https://{}:{}".format(serviceName,ip_port), data=data, extensions={"trace": log})
```

Or you can [inspect the network stream on the response instance](https://www.encode.io/httpcore/extensions/#extra-network-information)...

```python
resp.extensions["network_stream"].get_extra_info("ssl_object")
```

**Comment by karpetrosyan** [comment]
This solution may work if you use `http11` because the `network_stream` extension is not currently supported by `http2`.

```python
import httpx

conn = httpx.Client()
with conn.stream('GET', "https://www.google.com") as resp:
    stream = resp.extensions['network_stream']
    print(stream.get_extra_info('socket').cipher())
```
OUTPUT
```
('TLS_AES_256_GCM_SHA384', 'TLSv1.3', 256)
```
